packages/auth/src/core/credentials/email.test.ts

```diff
@@ -197,17 +197,9 @@ describe('core/credentials/email', () => {
           const stub = sinon.stub(recaptcha.enterprise, 'execute');
 
           // First verification should fail with 'wrong-site-key'
-          stub
-            .withArgs('wrong-site-key', {
-              action: RecaptchaActionName.SIGN_IN_WITH_PASSWORD
-            })
-            .rejects();
           // Second verifcation should succeed with site key refreshed
-          stub
-            .withArgs('site-key', {
-              action: RecaptchaActionName.SIGN_IN_WITH_PASSWORD
-            })
-            .returns(Promise.resolve('recaptcha-response'));
+          stub.onCall(0).rejects();
```

Comment:
I guess this isn't realistic since execute will never fail..

1. can you fix the comment on line 199 `First verification should fail with 'wrong-site-key'`?
2. Can we check that GET_RECAPTCHA_CONFIG endpoint got called twice? Thats what we are really testing, right?

Reply:
We're actually testing the L163-L167 here https://github.com/firebase/firebase-js-sdk/blob/f5aec7bd80a0adccc14088e6ff880d3f7627de5f/packages/auth/src/platform_browser/recaptcha/recaptcha_enterprise_verifier.ts#L163

The retry is less likely to happen since we return the fake token if the token is not available, while the verification can still fail for other reasons like the js script loading somehow fails. Here we force the first verification call to fail by making the first call to `execute` fails.

Reply:
Gotcha. it would be nice to check that there are 2 calls to get recaptcha config, to test that config refresh happened. 
> while the verification can still fail for other reasons like the js script loading somehow fails. 

True, but here we are stubbing execute(), not verify(), hence a little non-obvious. Maybe add a comment like "simulate verifier.verify() failure by simulating an execute() failure", if you agree that it might be clearer?

Reply:
Added check that there are 2 calls to get recaptcha config, and check the the site key is indeed updated.

Simulate the verify() failure by not setting window.grecaptcha(instead of fail execute()) to simplify the logic.